pkg/cloudprovider/aws/instancetype.go

```diff
@@ -43,10 +43,10 @@ type InstanceType struct {
 	provider           *v1alpha1.AWS
 }
 
-func newInstanceType(info ec2.InstanceTypeInfo, provider *v1alpha1.AWS) *InstanceType {
+func newInstanceType(info ec2.InstanceTypeInfo, includePodENI bool, provider *v1alpha1.AWS) *InstanceType {
```

Comment:
IMO we should pass context down and extract from there.

Reply:
Not much difference to me, this way is more explicit about what is being used. Technically using that for the context is a misuse of context per the Go docs since it's not request scoped:

> Use context values only for request-scoped data that transits processes and API boundaries, not for passing optional parameters to functions.

Reply:
I think this may have been written without deep thought towards usage of go. Controllers may not have been the initial use case and the language is very server specific. Given that we're already storing options in context, I think we might have already violated this rule, if we think it's the correct interpretation. 

That said, I'm not fussed either way.